omniledger/service/transaction.go

```diff
@@ -36,41 +43,17 @@ func NewNonce(buf []byte) Nonce {
 	return n
 }
 
-// NewInstanceID returns a new InstanceID given a slice of bytes. If the length
-// of the slice is not 64 bytes, it will return an all zero InstanceID.
-func NewInstanceID(buf []byte) InstanceID {
-	if len(buf) != 64 {
-		return InstanceID{zeroDarc, SubID{}}
-	}
-	return InstanceID{buf[0:32], NewSubID(buf[32:64])}
-}
-
-// Slice returns concatenated DarcID and InstanceID.
-func (iID InstanceID) Slice() []byte {
-	var out []byte
-	out = append(out, iID.DarcID[:]...)
-	return append(out, iID.SubID[:]...)
+// InstanceIDFromSlice converts the first 32 bytes of in into
+// an InstanceID.
+func InstanceIDFromSlice(in []byte) InstanceID {
+	var i InstanceID
+	copy(i[:], in)
+	return i
 }
 
-// Equal returns if both InstanceIDs point to the same instance.
-func (iID InstanceID) Equal(other InstanceID) bool {
```

Comment:
why remove `Equal` here? Most of our `*ID` do have `Equal`, which is convenient.